robot/impl/robot_reconfigure_test.go

```diff
@@ -1709,959 +3303,65 @@ func TestRobotReconfigure(t *testing.T) {
-				Name:       weak1Name.Name,
-				API:        weakAPI,
-				Model:      weakModel,
-				Attributes: rutils.AttributeMap{"version": 2},
-				ConvertedAttributes: &someTypeWithWeakAndStrongDepsConfig{
-					weakDeps: []resource.Name{base1Name},
-				},
-			},
-			{
-				Name:  base1Name.Name,
-				API:   base.API,
-				Model: fake.Model,
-			},
-			{
-				Name:  base2Name.Name,
-				API:   base.API,
-				Model: fake.Model,
-			},
-		},
-	}
-	test.That(t, weakCfg6.Ensure(false, logger), test.ShouldBeNil)
-	robot.Reconfigure(context.Background(), &weakCfg6)
-	res, err = robot.ResourceByName(weak1Name)
-	test.That(t, err, test.ShouldBeNil)
-	weak1, err = resource.AsType[*someTypeWithWeakAndStrongDeps](res)
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, weak1.resources, test.ShouldHaveLength, 2)
-	test.That(t, weak1.resources, test.ShouldContainKey, base1Name)
-	test.That(t, weak1.resources, test.ShouldContainKey, base2Name)
-
-	weakCfg7 := config.Config{
-		Components: []resource.Config{
-			{
-				Name:       weak1Name.Name,
-				API:        weakAPI,
-				Model:      weakModel,
-				Attributes: rutils.AttributeMap{"version": 3},
-				ConvertedAttributes: &someTypeWithWeakAndStrongDepsConfig{
-					deps:     []resource.Name{base2Name},
-					weakDeps: []resource.Name{base1Name},
-				},
-			},
-			{
-				Name:  base1Name.Name,
-				API:   base.API,
-				Model: fake.Model,
-			},
-			{
-				Name:  base2Name.Name,
-				API:   base.API,
-				Model: fake.Model,
-			},
-		},
-	}
-	test.That(t, weakCfg7.Ensure(false, logger), test.ShouldBeNil)
-	robot.Reconfigure(context.Background(), &weakCfg7)
-
-	res, err = robot.ResourceByName(weak1Name)
-	test.That(t, err, test.ShouldBeNil)
-	weak1, err = resource.AsType[*someTypeWithWeakAndStrongDeps](res)
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, weak1.resources, test.ShouldHaveLength, 2)
-	test.That(t, weak1.resources, test.ShouldContainKey, base1Name)
-	test.That(t, weak1.resources, test.ShouldContainKey, base2Name)
-}
-
-func TestDefaultServiceReconfigure(t *testing.T) {
-	logger := logging.NewTestLogger(t)
-
-	motionName := "motion"
-	cfg1 := &config.Config{
-		Services: []resource.Config{
-			{
-				Name:  motionName,
-				API:   motion.API,
-				Model: resource.DefaultServiceModel,
-			},
-		},
-	}
-	robot := setupLocalRobot(t, context.Background(), cfg1, logger)
-
-	rdktestutils.VerifySameResourceNames(t, robot.ResourceNames(),
-		[]resource.Name{
-			motion.Named(motionName),
-			sensors.Named(resource.DefaultServiceName),
-		},
-	)
-	sName := "sensors"
-	cfg2 := &config.Config{
-		Services: []resource.Config{
-			{
-				Name:  sName,
-				API:   sensors.API,
-				Model: resource.DefaultServiceModel,
-			},
-		},
-	}
-	robot.Reconfigure(context.Background(), cfg2)
-	rdktestutils.VerifySameResourceNames(
-		t,
-		robot.ResourceNames(),
-		[]resource.Name{
-			motion.Named(resource.DefaultServiceName),
-			sensors.Named(sName),
-		},
-	)
-}
-
-func TestStatusServiceUpdate(t *testing.T) {
```

Comment:
I removed this test altogether because the status service is deprecated.

Reply:
but we haven't removed the service itself yet, can you add it back?

Reply:
https://github.com/viamrobotics/rdk/pull/4138/commits/c19dccb6c1614c9af89d53354bc6b892867b0f1a